src/box/lua/iproto.c

```diff
@@ -28,240 +28,119 @@
 push_iproto_flag_constants(struct lua_State *L)
 {
-	const struct iproto_constant flags[] = {
-		{"COMMIT", IPROTO_FLAG_COMMIT},
-		{"WAIT_SYNC", IPROTO_FLAG_WAIT_SYNC},
-		{"WAIT_ACK",  IPROTO_FLAG_WAIT_ACK},
-	};
-	push_iproto_constant_subnamespace(L, "flag", flags, lengthof(flags));
+	push_iproto_constant_subnamespace(
+		L, "flag", iproto_flag_constants, iproto_flag_constants_size,
+		"IPROTO_FLAG_"
+	);
 }
 
 /**
- * Pushes IPROTO constants from `iproto_key` enumeration onto Lua stack.
+ * Pushes IPROTO constants generated from `IPROTO_KEYS` onto Lua stack.
  */
 static void
 push_iproto_key_enum(struct lua_State *L)
 {
-	const struct iproto_constant keys[] = {
-		{"REQUEST_TYPE", IPROTO_REQUEST_TYPE},
-		{"SYNC", IPROTO_SYNC},
-		{"REPLICA_ID", IPROTO_REPLICA_ID},
-		{"LSN", IPROTO_LSN},
-		{"TIMESTAMP", IPROTO_TIMESTAMP},
-		{"SCHEMA_VERSION", IPROTO_SCHEMA_VERSION},
-		{"SERVER_VERSION", IPROTO_SERVER_VERSION},
-		{"GROUP_ID", IPROTO_GROUP_ID},
-		{"TSN", IPROTO_TSN},
-		{"FLAGS", IPROTO_FLAGS},
-		{"STREAM_ID", IPROTO_STREAM_ID},
-		{"SPACE_ID", IPROTO_SPACE_ID},
-		{"INDEX_ID", IPROTO_INDEX_ID},
-		{"LIMIT", IPROTO_LIMIT},
-		{"OFFSET", IPROTO_OFFSET},
-		{"ITERATOR", IPROTO_ITERATOR},
-		{"INDEX_BASE", IPROTO_INDEX_BASE},
-		{"FETCH_POSITION", IPROTO_FETCH_POSITION},
-		{"KEY", IPROTO_KEY},
-		{"TUPLE", IPROTO_TUPLE},
-		{"FUNCTION_NAME", IPROTO_FUNCTION_NAME},
-		{"USER_NAME", IPROTO_USER_NAME},
-		{"INSTANCE_UUID", IPROTO_INSTANCE_UUID},
-		{"REPLICASET_UUID", IPROTO_REPLICASET_UUID},
-		{"VCLOCK", IPROTO_VCLOCK},
-		{"EXPR", IPROTO_EXPR},
-		{"OPS", IPROTO_OPS},
-		{"BALLOT", IPROTO_BALLOT},
-		{"TUPLE_META", IPROTO_TUPLE_META},
-		{"OPTIONS", IPROTO_OPTIONS},
-		{"OLD_TUPLE", IPROTO_OLD_TUPLE},
-		{"NEW_TUPLE", IPROTO_NEW_TUPLE},
-		{"IPROTO_AFTER_POSITION", IPROTO_AFTER_POSITION},
-		{"IPROTO_AFTER_TUPLE", IPROTO_AFTER_TUPLE},
-		{"DATA", IPROTO_DATA},
-		{"ERROR_24", IPROTO_ERROR_24},
-		{"METADATA", IPROTO_METADATA},
-		{"BIND_METADATA", IPROTO_BIND_METADATA},
-		{"BIND_COUNT", IPROTO_BIND_COUNT},
-		{"IPROTO_POSITION", IPROTO_POSITION},
-		{"SQL_TEXT", IPROTO_SQL_TEXT},
-		{"SQL_BIND", IPROTO_SQL_BIND},
-		{"SQL_INFO", IPROTO_SQL_INFO},
-		{"STMT_ID", IPROTO_STMT_ID},
-		{"REPLICA_ANON", IPROTO_REPLICA_ANON},
-		{"ID_FILTER", IPROTO_ID_FILTER},
-		{"ERROR", IPROTO_ERROR},
-		{"TERM", IPROTO_TERM},
-		{"VERSION", IPROTO_VERSION},
-		{"FEATURES", IPROTO_FEATURES},
-		{"TIMEOUT", IPROTO_TIMEOUT},
-		{"EVENT_KEY", IPROTO_EVENT_KEY},
-		{"EVENT_DATA", IPROTO_EVENT_DATA},
-		{"TXN_ISOLATION", IPROTO_TXN_ISOLATION},
-		{"VCLOCK_SYNC", IPROTO_VCLOCK_SYNC},
-	};
-	push_iproto_constant_subnamespace(L, "key", keys, lengthof(keys));
-	for (size_t i = 0; i < lengthof(keys); ++i) {
-		size_t len = strlen(keys[i].name);
-		char *lowercase = strtolowerdup(keys[i].name);
+	push_iproto_constant_subnamespace(
+		L, "key", iproto_key_constants, iproto_key_constants_size,
+		"IPROTO_"
+	);
+	const size_t prefix_size = sizeof("IPROTO_") - 1;
```

Comment:
Nit: this is not considered an integer constant expression in C — better just inline it.

Reply:
Why isn't it an integer constant expression?

Reply:
Please, look at the specification of [integer constant expressions](https://en.cppreference.com/w/c/language/constant_expression) in C.

Reply:
BTW, please don't use `sizeof` for string literals (see my comment below about optimization of `strlen` on string literals).

Reply:
Okay, fixed.